### ISSUE TITLE
[StableHLO] Transpose op failure

### ISSUE THREAD

**Issue by mmanzoorTT** [tenstorrent/tt-mlir#849]
StableHLO to TTIR conversion fails to legalize `transpose` op for 3 and higher dimensional tensors. A sample test case can be found in [tt-xla PR](https://github.com/tenstorrent/tt-xla/pull/15)

**Comment by sdjordjevicTT** [comment]
Is this a duplicate of this?
https://github.com/tenstorrent/tt-mlir/issues/558

We filed an issue on metal regarding the support of -3 dim on transpose, and they are working on it:
https://github.com/tenstorrent/tt-metal/issues/13005

**Comment by mmanzoorTT** [comment]
This issue depends on #558 (not a duplicate). We need to fix stableHLO to TTIR conversion in `StableHLOToTTIRPatterns.cpp` as well once #558 is resolved.

**Comment by sdjordjevicTT** [comment]
Oh, I get it, thanks!

**Comment by skhatriTT** [comment]
Hi @mmanzoorTT I noticed that the issue [#558](https://github.com/tenstorrent/tt-mlir/issues/558) has been resolved. What is our status on this?

**Comment by mmanzoorTT** [comment]
Hi @skhatriTT yes, this issue is resolved.
